pkg/assets/rbac.go

```diff
@@ -42,19 +42,46 @@ func (crb *clusterRoleBindingApplier) New(kubeconfigPath string) {
 	crb.client = kubernetes.NewForConfigOrDie(rest.AddUserAgent(restConfig, "rbac-agent"))
 }
 
-func (crb *clusterRoleBindingApplier) Reader(objBytes []byte, _ RenderFunc, _ RenderParams) {
+func (crb *clusterRoleBindingApplier) Read(objBytes []byte, _ RenderFunc, _ RenderParams) {
 	obj, err := runtime.Decode(rbacCodecs.UniversalDecoder(rbacv1.SchemeGroupVersion), objBytes)
 	if err != nil {
 		panic(err)
 	}
 	crb.crb = obj.(*rbacv1.ClusterRoleBinding)
 }
 
-func (crb *clusterRoleBindingApplier) Applier(ctx context.Context) error {
+func (crb *clusterRoleBindingApplier) Handle(ctx context.Context) error {
 	_, _, err := resourceapply.ApplyClusterRoleBinding(ctx, crb.client.RbacV1(), assetsEventRecorder, crb.crb)
 	return err
 }
 
+type clusterRoleBindingDeleter struct {
+	client *kubernetes.Clientset
+	crb    *rbacv1.ClusterRoleBinding
+}
+
+func (crb *clusterRoleBindingDeleter) New(kubeconfigPath string) {
+	restConfig, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		panic(err)
+	}
+
+	crb.client = kubernetes.NewForConfigOrDie(rest.AddUserAgent(restConfig, "rbac-agent"))
+}
+
+func (crb *clusterRoleBindingDeleter) Read(objBytes []byte, _ RenderFunc, _ RenderParams) {
+	obj, err := runtime.Decode(rbacCodecs.UniversalDecoder(rbacv1.SchemeGroupVersion), objBytes)
+	if err != nil {
+		panic(err)
```

Comment:
Can we avoid relying on calling panic? By the time some of these methods are invoked other services and processes are running and it would be better to stop them gracefully.

Reply:
Yeah, I noticed a few and then commented on 1 of them. Are all of the panics new in this PR, or were we calling panic before and now just call it from more places?

Reply:
The PR had one panic per `Reader` function before, and most of those are now generalized into a single function so they are not present anymore. This function (and the remaining panics) can not use the generic function because it requires a different reading method.
I had the intention of doing further changes within this package, but they exceed the intention of this PR.

Reply:
OK, please file a ticket to remove the panics as a separate PR.